framework/include/csg/CSGBase.h

```diff
@@ -275,6 +291,301 @@ class CSGBase
+   *
+   * @param lattice pointer to lattice to add
+   *
+   * @return reference to CSGLattice that was added
+   */
+  const CSGLattice & addLattice(std::unique_ptr<CSGLattice> lattice);
+
+  /**
+   * @brief set location in the lattice to be the provided universe
+   *
+   * @param lattice lattice to update
+   * @param universe universe to set at the location
+   * @param index index of the lattice element (int, int)
+   */
+  void setUniverseAtLatticeIndex(const CSGLattice & lattice,
+                                 const CSGUniverse & universe,
+                                 std::pair<int, int> index);
+
+  /**
+   * @brief Set provided universes as the layout of the lattice.
+   *
+   * @param lattice lattice to add universes to
+   * @param universes list of list of universes in the proper layout for the lattice type and
+   * dimensions
+   */
+  void setLatticeUniverses(
+      const CSGLattice & lattice,
+      std::vector<std::vector<std::reference_wrapper<const CSGUniverse>>> & universes);
+
+  /**
+   * @brief rename the lattice
+   *
+   * @param lattice lattice to rename
+   * @param name new name
+   */
+  void renameLattice(const CSGLattice & lattice, const std::string & name)
+  {
+    _lattice_list.renameLattice(lattice, name);
+  }
+
+  /**
+   * @brief Set the outer fill for the lattice to the material name provided. This will set the
+   * outer type to CSG_MATERIAL regardless of its previous outer type.
+   *
+   * @param lattice lattice to update
+   * @param outer_name name of material to use as outer fill between lattice elements
+   */
+  void setLatticeOuter(const CSGLattice & lattice, const std::string & outer_name);
+
+  /**
+   * @brief Set the outer fill for the lattice to the universe provided. This will set the outer
+   * type to UNIVERSE regardless of its previous outer type.
+   *
+   * @param lattice lattice to update
+   * @param outer_univ universe to use as outer fill between lattice elements
+   */
+  void setLatticeOuter(const CSGLattice & lattice, const CSGUniverse & outer_univ);
+
+  /**
+   * @brief reset the outer fill for the lattice to VOID
+   *
+   * @param lattice lattice to update
+   */
+  void resetLatticeOuter(const CSGLattice & lattice);
+
+  /**
+   * @brief Get all lattice objects
+   *
+   * @return list of references to CSGLattice objects in this CSGBase instance
+   */
+  std::vector<std::reference_wrapper<const CSGLattice>> getAllLattices() const
+  {
+    return _lattice_list.getAllLattices();
+  }
+
+  /**
+   * @brief Get a lattice object of the specified type by name
+   * This is a templated method with a default type of CSGLattice. If a specific lattice type
+   * is needed, it can be specified when calling. If the type is unknown or not specified,
+   * it will default to CSGLattice to get the base class reference.
+   * NOTE: if CSGLattice is used as the template type, any lattice type-specific attributes or
+   * methods may not be accessible.
```

Comment:
```suggestion
   * methods may not be accessible, except using a reference cast.
```
or delete the advice? This is like a C++/typed language rule